herald/herald/Sensor/BLE/BLEDatabase.swift

```diff
@@ -320,38 +320,59 @@ class BLEPseudoDeviceAddress {
         }}
     
     init?(fromAdvertisementData: [String: Any]) {
-        guard let manufacturerData = fromAdvertisementData["kCBAdvDataManufacturerData"] as? Data else {
+        guard let manufacturerData = fromAdvertisementData[CBAdvertisementDataManufacturerDataKey] as? Data else {
             return nil
         }
-        guard let manufacturerId = manufacturerData.uint16(0), manufacturerId == BLESensorConfiguration.manufacturerIdForSensor else {
+        guard let manufacturerId = manufacturerData.uint16(0) else {
             return nil
         }
-        guard manufacturerData.count == 8 else {
-            return nil
+        // HERALD pseudo device address
+        if manufacturerId == BLESensorConfiguration.manufacturerIdForSensor, manufacturerData.count == 8 {
+            data = Data(manufacturerData.subdata(in: 2..<8))
+            var longValueData = Data(repeating: 0, count: 2)
+            longValueData.append(data)
+            guard let longValue = longValueData.int64(0) else {
+                return nil
+            }
+            address = Int64(longValue)
+        }
+        // Legacy pseudo device address
+        else if BLESensorConfiguration.interopOpenTraceEnabled, manufacturerId == BLESensorConfiguration.interopOpenTraceManufacturerId, manufacturerData.count > 2 {
+            data = Data(manufacturerData.subdata(in: 2..<min(8, manufacturerData.count)))
+            var longValueData = Data(data)
+            if longValueData.count < 8 {
+                longValueData.append(Data(repeating: 0, count: 8 - longValueData.count))
```

Comment:
This is appended for openTrace but prepended for Herald. Is this correct?

Reply:
Doesn't matter either way in this instance because OpenTrace address is being used internally only. Have updated to make it consistent.